kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/templates/testclients/TestClientsJobTemplates.java

```diff
@@ -51,25 +57,265 @@ public static JobBuilder defaultAdminClientJob(String jobName, List<String> args
+                        .withEnv(kafkaGoProducerEnvVars(bootstrap, topicName))
+                        .withSecurityContext(jobsSecurityContext())
+                        .build())
+                .endSpec()
+                .endTemplate()
+                .endSpec();
+    }
+
+    /**
+     * Default kafka go consumer job builder.
+     *
+     * @param jobName the job name
+     * @param bootstrap the bootstrap
+     * @param topicName the topic name
+     * @return the job builder
+     */
+    public static JobBuilder defaultKafkaGoConsumerJob(String jobName, String bootstrap, String topicName) {
+        return baseClientJob(jobName)
+                .editSpec()
+                .editTemplate()
+                .editSpec()
+                .withContainers(new ContainerBuilder()
+                        .withName("kafka-go-consumer")
+                        .withImage("ppatierno/kafka-go-consumer:latest")
+                        .withImagePullPolicy(Constants.PULL_IMAGE_IF_NOT_PRESENT)
+                        .withEnv(kafkaGoConsumerEnvVars(bootstrap, topicName))
+                        .withSecurityContext(jobsSecurityContext())
+                        .build())
+                .endSpec()
+                .endTemplate()
+                .endSpec();
+    }
+
+    private static List<EnvVar> testClientsProducerEnvVars(String bootstrap, String topicName, int numOfMessages, String message) {
+        List<EnvVar> envVarList = new ArrayList<>();
+        envVarList.add(new EnvVarBuilder()
+                .withName(BOOTSTRAP_VAR)
+                .withValue(bootstrap)
+                .build());
+        envVarList.add(new EnvVarBuilder()
+                .withName(DELAY_MS_VAR)
+                .withValue("1000")
+                .build());
+        envVarList.add(new EnvVarBuilder()
+                .withName(TOPIC_VAR)
+                .withValue(topicName)
+                .build());
+        envVarList.add(new EnvVarBuilder()
+                .withName(MESSAGE_COUNT_VAR)
+                .withValue(String.valueOf(numOfMessages))
+                .build());
+        envVarList.add(new EnvVarBuilder()
+                .withName(MESSAGE_VAR)
+                .withValue(message)
+                .build());
+        envVarList.add(new EnvVarBuilder()
+                .withName(PRODUCER_ACKS_VAR)
+                .withValue("all")
+                .build());
+        envVarList.add(new EnvVarBuilder()
+                .withName(LOG_LEVEL_VAR)
+                .withValue("INFO")
+                .build());
+        envVarList.add(new EnvVarBuilder()
+                .withName(CLIENT_TYPE_VAR)
+                .withValue("KafkaProducer")
+                .build());
+
+        return envVarList;
+    }
+
+    private static List<EnvVar> testClientsConsumerEnvVars(String bootstrap, String topicName, int numOfMessages) {
+        List<EnvVar> envVarList = new ArrayList<>();
+        envVarList.add(new EnvVarBuilder()
```

Comment:
suggest it's worth extracting an `EnvVar envVar(String name, String value)` method

Reply:
Yeah and couple that with `List.of(envVar(NAME_1, VALUE_1), envVar(NAME_2, VALUE_2), ... )`